mavros/include/mavros/mavros_uas.h

```diff
@@ -310,8 +313,43 @@ class UAS {
 
 	/* -*- autopilot version -*- */
 	uint64_t get_capabilities();
+
+	/**
+	 * @brief Function to check if the flight controller has a capability
+	 *
+	 * @param capabilities can accept a multiple capability params either in enum or int from
+	 */
+	template<typename T>
+	bool has_capability(T capability){
+		return static_cast<uint64_t>(fcu_capabilities) & static_cast<uint64_t>(capability);
```

Comment:
Because it's atomic uint64_t better first to copy it's state, then do OR.
Also maybe better to use `enum_value()` for capability cast.

Reply:
Probably you should use `get_capabilities()` here.

Reply:
Ok, I'll use get_capabilities(). The reason I didn't use enum_value() is that underlying_type on a non-enum value is undefined behavior. I wrote the templated code so that either the int code or the Enum value could be used to check the capabilities, should I only accept enum values? This would remove the need for a  template I belive which might speed up the compile time and force people not to use magic numbers.

Reply:
I am using static_cast instead of enum_value since the template can also accept ints and underlying_type on non-enum values is undefined behavior. Thinking about it though I did switch to enum_value and added a static assert to give a warning message if a non enum type is used so users cant use a magic number.